### PR TITLE
split player::disp_info() into constituent parts

### PR DESCRIPTION
**Pull request by KorGgenT**
#### Summary

```SUMMARY: Infrastructure "Split player::disp_info() into constituent parts"```
#### Purpose of change

this function was 1200 lines long.
#### Describe the solution

I have created individual static functions f each of the 6 windows within the switch statement
- draw_stats_tab
- draw_encumbrance_tab
- draw_traits_tab
- draw_bionics_tab
- draw_effects_tab
- draw_skills_tab
I have also created two other functions that have large chunks of code
- draw_grid_borders
- draw_initial_windows

The only thing I've done is created static functions and copy pasted the bodies into the new functions, and named the parameters the same as the variables they were using. as a result, most of these have references passed into these functions. 

That being said, there is no change of how these work in game: the character menus of both the player and npcs.

#### Describe alternatives you've considered

deleting the whole thing and running away screaming
#### Additional context

i left it in player:: because there are a couple player only functions. this needs to eventually be migrated into Character because this is also used to view NPC stats.

### CODE REVIEW

File: src/player_display.cpp [hunk 1]
```diff
@@ -237,163 +246,511 @@ static bool is_cqb_skill( const skill_id &id )
...
-            starvation_text << _( "Strength" ) << " -" << static_cast<int>( starvation_base_penalty / 500 ) <<
-                            "   ";
-            if( starvation_base_penalty > 1000 ) {
-                starvation_text << _( "Dexterity" ) << " -" << static_cast<int>( starvation_base_penalty / 1000 ) <<
-                                "   ";
-                starvation_text << _( "Intelligence" ) << " -" << static_cast<int>( starvation_base_penalty / 1000 )
-                                << "   ";
-            }
+static void draw_effects_tab( const catacurses::window &w_effects, const catacurses::window &w_info,
+                              player &you, unsigned int &line, int &curtab, input_context &ctxt, bool &done,
```
Comment (kevingranade) [inline src/player_display.cpp]:
```suggestion
                              unsigned int &line, int &curtab, input_context &ctxt, bool &done,
```

File: src/player_display.cpp [hunk 2]
```diff
@@ -785,474 +1312,31 @@ Strength - 4;    Dexterity - 4;    Intelligence - 4;    Perception - 4" ) );
...
-                        trim_and_print( w_effects, static_cast<int>( i ) + 1, 0, getmaxx( w_effects ) - 1, c_light_gray,
-                                        effect_name[i] );
-                    }
-                    wrefresh( w_effects );
-                    line = 0;
-                    curtab = action == "NEXT_TAB" ? 1 : curtab - 1;
-                } else if( action == "QUIT" ) {
-                    done = true;
-                }
+                draw_effects_tab( w_effects, w_info, *this, line, curtab, ctxt, done, action,
```
Comment (kevingranade) [inline src/player_display.cpp]:
```suggestion
                draw_effects_tab( w_effects, w_info, line, curtab, ctxt, done, action,
```

**Comment by KorGgenT** [comment]
jenkins rebuild

**Comment by mlangsdorf** [comment]
I'd like to see it reorganized into a class that owns the windows, but that can be done in a future PR.

**Pull request merged**